systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -639,6 +642,154 @@ void testKafkaRebalanceAutoApprovalMechanism() {
+            @Step(value = "Confirm that the cluster is stable after scaling operations.", expected = "Cluster returns to a stable state with initial number of brokers and Cruise Control completed the rebalancing.")
+        },
+        labels = {
+            @Label(value = TestDocsLabels.CRUISE_CONTROL),
+        }
+    )
+    void testAutoKafkaRebalanceScaleUpScaleDown() {
+        final TestStorage testStorage = new TestStorage(ResourceManager.getTestContext());
+        final String scaleUpKafkaRebalanceTemplateName = testStorage.getClusterName() + "-scale-up";
+        final String scaleDownKafkaRebalanceTemplateName = testStorage.getClusterName() + "-scale-down";
+        final int initialReplicas = 3;
+
+        resourceManager.createResourceWithWait(
+            NodePoolsConverter.convertNodePoolsIfNeeded(
+                KafkaNodePoolTemplates.brokerPool(testStorage.getNamespaceName(), testStorage.getBrokerPoolName(), testStorage.getClusterName(), initialReplicas).build(),
+                KafkaNodePoolTemplates.controllerPool(testStorage.getNamespaceName(), testStorage.getControllerPoolName(), testStorage.getClusterName(), initialReplicas).build()
+            )
+        );
+
+        // create two KafkaRebalance with marking resource as configuration template
+        resourceManager.createResourceWithoutWait(
+            KafkaRebalanceTemplates.kafkaRebalance(testStorage.getNamespaceName(), testStorage.getClusterName())
+                .editMetadata()
+                    .addToAnnotations(Annotations.ANNO_STRIMZI_IO_REBALANCE, "template")
+                    .withName(scaleUpKafkaRebalanceTemplateName)
+                .endMetadata()
+                .build(),
+            KafkaRebalanceTemplates.kafkaRebalance(testStorage.getNamespaceName(), testStorage.getClusterName())
+                .editMetadata()
+                    .addToAnnotations(Annotations.ANNO_STRIMZI_IO_REBALANCE, "template")
+                    .withName(scaleDownKafkaRebalanceTemplateName)
+                .endMetadata()
+                .build()
+        );
+
+        // define CC with two auto-rebalance modes (add/remove) and corresponding templates
+        resourceManager.createResourceWithWait(
+            KafkaTemplates.kafkaWithCruiseControlTunedForFastModelGeneration(testStorage.getNamespaceName(), testStorage.getClusterName(), 3, 3)
+                .editSpec()
+                    .editCruiseControl()
+                        .withAutoRebalance(
+                            new KafkaAutoRebalanceConfigurationBuilder()
+                                .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                                .withNewTemplate(scaleUpKafkaRebalanceTemplateName)
+                                .build(),
+                            new KafkaAutoRebalanceConfigurationBuilder()
+                                .withMode(KafkaRebalanceMode.REMOVE_BROKERS)
```

Comment:
This will be likely broken by #10744?

Reply:
Yes it will be. At this point I think it's better waiting for #10744 to be merged (regression running) and then rebasing this one.